dipy/align/tests/test_imaffine.py

```diff
@@ -312,7 +320,232 @@ def test_mi_gradient():
+    nx = dom_shape[0]
+    ny = dom_shape[1]
+    nz = dom_shape[2]
+    # Radius of the circle/sphere (testing image)
+    radius = 16
+    # Rotation axis (used for 3D transforms only)
+    rot_axis = np.array([.5, 2.0, 1.5])
+    # Arbitrary transform parameters
+    t = 0.15
+    rotations = [-1 * np.pi / 10.0, 0.0, np.pi / 10.0]
+    scales = [0.9,  1.0, 1.1]
+    for dim in [2, 3]:
+        # Setup current dimension
+        if dim == 2:
+            # Create image of a circle
+            img = vf.create_circle(cod_shape[0], cod_shape[1], radius)
+            oracle_linear = vf.transform_2d_affine
+            oracle_nn = vf.transform_2d_affine_nn
+        else:
+            # Create image of a sphere
+            img = vf.create_sphere(cod_shape[0], cod_shape[1], cod_shape[2],
+                                   radius)
+            oracle_linear = vf.transform_3d_affine
+            oracle_nn = vf.transform_3d_affine_nn
+        img = np.array(img)
+        # Translation is the only parameter differing for 2D and 3D
+        translations = [t * dom_shape[:dim]]
+        # Generate affine transforms
+        gt_affines = create_affine_transforms(dim, translations, rotations,
+                                              scales, rot_axis)
+        # Include the None case
+        gt_affines.append(None)
+
+        for affine in gt_affines:
+
+            # make both domain point to the same physical region
+            # It's ok to use the same transform, we just want to test
+            # that this information is actually being considered
+            domain_grid2world = affine
+            codomain_grid2world = affine
+            grid2grid_transform = affine
+
+            # Evaluate the transform with vector_fields module (already tested)
+            expected_linear = oracle_linear(img, dom_shape[:dim],
+                                            grid2grid_transform)
+            expected_nn = oracle_nn(img, dom_shape[:dim], grid2grid_transform)
+
+            # Evaluate the transform with the implementation under test
+            affine_map = imaffine.AffineMap(affine,
+                                            dom_shape[:dim], domain_grid2world,
+                                            cod_shape[:dim], codomain_grid2world)
+            actual_linear = affine_map.transform(img, interp='linear')
+            actual_nn = affine_map.transform(img, interp='nearest')
+            assert_array_almost_equal(actual_linear, expected_linear)
+            assert_array_almost_equal(actual_nn, expected_nn)
+
+            # Test set_affine with valid matrix
+            affine_map.set_affine(affine)
+            if affine is None:
+                assert(affine_map.affine is None)
+                assert(affine_map.affine_inv is None)
+            else:
+                assert_array_equal(affine, affine_map.affine)
+                actual = affine_map.affine.dot(affine_map.affine_inv)
```

Comment:
This is just to show that these properties really are the inverses of each other?